### PR TITLE
Add reverse rewrite and inline rules

### PR DESCRIPTION
**Pull request by ericmj**

### CODE REVIEW

File: lib/elixir/src/elixir_rewrite.erl [hunk 1]
```diff
@@ -20,218 +21,245 @@
...
+  case inner_inline(erl_to_ex, Mod, Fun, length(Args)) of
+    false -> inner_rewrite(erl_to_ex, Mod, Fun, Args);
+    {ExMod, ExFun} -> {ExMod, ExFun, Args}
+  end.
+
+%% Inline  rules
+%%
 %% Inline rules are straightforward, they keep the same
 %% number and order of arguments and show up on captures.
+inline(Mod, Fun, Args) -> inner_inline(ex_to_erl, Mod, Fun, Args).
```
Comment (josevalim) [inline lib/elixir/src/elixir_rewrite.erl]:
```suggestion
inline(Mod, Fun, Arity) -> inner_inline(ex_to_erl, Mod, Fun, Arity).
```

?

File: lib/elixir/src/elixir_rewrite.erl [hunk 2]
```diff
@@ -20,218 +21,245 @@
...
+inner_rewrite(ex_to_erl, ?tuple, insert_at, [Tuple, Index, Term]) ->
   {erlang, insert_element, [increment(Index), Tuple, Term]};
-rewrite(Receiver, Fun, Args) ->
-  {Receiver, Fun, Args}.
+inner_rewrite(ex_to_erl, ?kernel, elem, [Tuple, Index]) ->
+  {erlang, element, [increment(Index), Tuple]};
+inner_rewrite(ex_to_erl, ?kernel, put_elem, [Tuple, Index, Value]) ->
+  {erlang, setelement, [increment(Index), Tuple, Value]};
+
+inner_rewrite(erl_to_erl, erlang, delete_element, [Index, Tuple]) when is_number(Index) ->
```
Comment (josevalim) [inline lib/elixir/src/elixir_rewrite.erl]: `erl_to_erl` are typos, right?

Comment (josevalim) [reply lib/elixir/src/elixir_rewrite.erl]: If so, we are probably missing some tests. :)

Comment (ericmj) [reply lib/elixir/src/elixir_rewrite.erl]: Advice on where to put the tests?

Comment (josevalim) [reply lib/elixir/src/elixir_rewrite.erl]: I would ideally check it on the module check error messages to see if they were properly reversed. Thoughts?

**Pull request merged**